Signum.Engine.Extensions/Files/FilePathEmbeddedLogic.cs

```diff
@@ -23,13 +23,22 @@ public static void Start(SchemaBuilder sb)
 
             FilePathEmbedded.OnPreSaving += efp =>
             {
-                if(efp.BinaryFile != null) //First time
+                if (efp.BinaryFile != null) //First time
                 {
                     var task = efp.SaveFileAsync();
                     Transaction.PreRealCommit += data =>
                     {
                         var a = efp; //For debugging
-                        task.Wait();
+                        if (task.Status == TaskStatus.WaitingForActivation)

```

Comment:
This is a hack. We need a better solution

Reply:
Temporarily removed to accept PR 👍